component/identity/src/main/java/org/exoplatform/services/organization/idm/GroupDAOImpl.java

```diff
@@ -152,6 +152,96 @@ public void addChild(Group parent, Group child, boolean broadcast) throws Except
+        try {
+            jbidParentTargetGroup = getIdentitySession().getPersistenceManager()
+                                                        .findGroup(plParentTargetGroupName,
+                                                                   orgService.getConfiguration().getGroupType(parentTargetGroup.getParentId()));
+        } catch (Exception e) {
+            // TODO:
+            handleException("Identity operation error: ", e);
+        }
+
+        //find groupToMove
+        org.picketlink.idm.api.Group jbidGroupToMove = null;
+        String plGroupToMoveName = getPLIDMGroupName(groupToMove.getGroupName());
+        try {
+            jbidGroupToMove = getIdentitySession().getPersistenceManager()
+                                                        .findGroup(plGroupToMoveName,
+                                                                   orgService.getConfiguration().getGroupType(groupToMove.getParentId()));
+        } catch (Exception e) {
+            // TODO:
+            handleException("Identity operation error: ", e);
+        }
+
+        //if one is miossing => error
+        if (jbidParentOriginGroup == null) {
+            throw new Exception("Group " + jbidParentOriginGroup + " does not exists");
+        }
+        if (jbidParentTargetGroup == null) {
+            throw new Exception("Group " + jbidParentOriginGroup + " does not exists");
+        }
+        if (jbidGroupToMove == null) {
+            throw new Exception("Group " + jbidGroupToMove + " does not exists");
+        }
+
+
+
+
+
+
+        //use RelationshiManager.disassociated
+        try {
+            getIdentitySession().getRelationshipManager()
+                                                 .disassociateGroups(jbidParentOriginGroup, List.of(jbidGroupToMove));
+        } catch (Exception e) {
+            handleException("Cannot dissociate: " + plGroupToMoveName + " to "+plParentOriginGroupName+"; ", e);
+        }
+        //use RelationshiManager.associate
+        try {
+            getIdentitySession().getRelationshipManager()
+                                .associateGroups(jbidParentTargetGroup,jbidGroupToMove);
+        } catch (Exception e) {
+            handleException("Cannot associate: " + plGroupToMoveName + " to "+plParentTargetGroupName+"; ", e);
+        }
+
+        //
+//
+//        org.picketlink.idm.api.Group jbidParentOriginGroup = null;
+//        String plParentOriginGroupName = getPLIDMGroupName(originGroup.getParentId());
+//        try {
+//            jbidOriginGroup = getIdentitySession().getPersistenceManager().findGroup(plParentOriginGroupName,
+//                                                                                     orgService.getConfiguration().getGroupType(originGroup.getParentId()));
+//        } catch (Exception e) {
+//            // TODO:
+//            handleException("Identity operation error: ", e);
+//        }
+//
+//        if (jbidOriginGroup == null) {
+//
+//        }
```

Comment:
COuld you remove this block if it is not needed

Reply:
done